src/Datadog.Trace/AsyncLocalCompat.cs

```diff
@@ -2,26 +2,22 @@ namespace Datadog.Trace
 {
 #if NET45
     using System;
+    using System.Runtime.Remoting;
     using System.Runtime.Remoting.Messaging;
 
-    // TODO:bertrand revisit this when we want to support multiple AppDomains
```

Comment:
Is it possible to do a unit test with multiple AppDomains? It would be nice to have a test for this.

Reply:
It's could be a little tricky, but I think we can do it be starting a separate process like we do for the integration tests. I created this [Trello card](https://trello.com/c/reZ3ATBe) to keep track of this.